core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -47,4 +48,10 @@ public CloseableIterable<CombinedScanTask> planTasks() {
     return TableScanUtil.planTasks(
         splitFiles, targetSplitSize(), splitLookback(), splitOpenFileCost());
   }
+
+  /** @deprecated will be removed in 1.3.0; use {@link SnapshotScan#scanMetrics()} instead. */
```

Comment:
Is there a reason for this ?  It will inherit automatically SnapshotScan.scanMetrics, wont it?

Reply:
ah yes you're totally right. I overlooked that the API break was actually because of switchting the method from `protected` to package-private. Making the method `protected` in the super class fixes this, so we don't need to deprecate anything here.